bookstore/archive.py

```diff
@@ -29,6 +45,46 @@ def __init__(self, *args, **kwargs):
             s3_additional_kwargs={},
         )
 
+        self._thread_pool = ThreadPoolExecutor(max_workers=30)
```

Comment:
Worth documenting the magic number or better yet making it configurable?

Reply:
Configurable is probably the right approach, as I stuck this in here just to have something to work with for the moment.

Reply:
This is now configurable.